### PR TITLE
Implement jaspReport functionality

### PR DESCRIPTION
**Pull request by JorisGoosen**
This belongs to a larger PR to jasp-desktop, it implements a jaspObject that an analysis can use to display warnings or important messages to the user.
The important messages will show at the top of the results.

JASP is also able to collect these reports from the entire resultswindow and write them to files that can then be easily served through a webserver. All of this functionality is hidden/disabled if reporting-mode is off. And reportingmode can only be enabled/disabled when running in development mode. (for now anyway, as this is really meant for users atm)

### CODE REVIEW

**Comment by JorisGoosen** [comment]
If @vandenman wants to have a look at my R changes Id appreciate it

File: src/jaspContainer.h [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #include "jaspState.h"
 #include "jaspHtml.h"
 #include "jaspQmlSource.h"
+#include "jaspReport.h"
```
Comment (vandenman) [inline src/jaspContainer.h:9]: deze wordt toch niet gebruikt?

**Review by vandenman — APPROVED** [review]

File: src/jaspContainer.cpp [hunk 2]
```diff
@@ -82,6 +83,7 @@ Rcpp::RObject jaspContainer::wrapJaspObject(jaspObject * ref)
 	case jaspObjectType::container:	return Rcpp::wrap(jaspContainer_Interface(ref));
 	case jaspObjectType::qmlSource:	return Rcpp::wrap(jaspQmlSource_Interface(ref));
 	case jaspObjectType::column:	return Rcpp::wrap(jaspColumn_Interface(ref));
+	case jaspObjectType::report:	return Rcpp::wrap(jaspReport_Interface(ref));
```
Comment (JorisGoosen) [inline src/jaspContainer.cpp:86]: hierzo @vandenman

File: src/jaspContainer.h (hunk 1 above)
Comment (JorisGoosen) [reply src/jaspContainer.h:9]: zie https://github.com/jasp-stats/jaspBase/pull/100#discussion_r1022825725 

maar ook zoiezo voor jaspResults moet de klasse ergens worden geinclude en aangezien results kind is van container is dit dus een prima plek.

**Review by boutinb — APPROVED** [review]

**Pull request merged**